client/my-sites/plan-features/header.jsx

```diff
@@ -143,23 +143,48 @@ export class PlanFeaturesHeader extends Component {
 	}
 
 	getDiscountTooltipMessage() {
-		const { currencyCode, currentSitePlan, translate, rawPrice } = this.props;
+		const { currencyCode, currentSitePlan, translate, rawPrice, discountPrice } = this.props;
+		const price = formatCurrency( rawPrice, currencyCode );
+		const isDiscounted = !! discountPrice;
```

Comment:
Same as : https://github.com/Automattic/wp-calypso/pull/45063/files#r485297150
Feel a bit uneasy about relying on coercion to arrive at this boolean.

Reply:
Happy to change it but would you mind explaining the reasoning behind this? !! is in use in many places in the codebase, I followed a pattern.

Reply:
Ya I am just starting to see this now. It is used everywhere. 
The reasoning is because discount price is a price and if you rely on coercion to decide this value for you suddenly it opens up multiple ways for this to be effected at least in the readers mind. 

But seeing as discounted price is a number the more readable way would be to come at a logical comparison like
```javascript
const isDiscounted = price > discountPrice;
```

Reply:
I don't quite like that the definition of a discount here would be having or not having a price of a certain kind - this approach somewhat represents how it works on the backend, and I can trace it back to old code that has nothing to do with what we have here, from the early days of the store, and the first discount coupons.

The other place where I used !! it ended up being what you mention - and I had to change it to `price > discountPrice` and it caused one of the bugs reported.